### PR TITLE
Radio group documentation update

### PR DESCRIPTION
**Pull request by GomezIvann**
Summary:

- Documentation update, more examples and further error handling explanation.
- Added component to the theme builder.
- The radio button component is marked as deprecated.
- New aria attribute added to the component: `aria-orientation`.
- Updated tests.
- Comments on types.

### CODE REVIEW

**Review by marcialfps — CHANGES_REQUESTED** [review]
Comments about tests:

- _Initial render has correct aria attributes and tabIndex_ case. Add `aria-live` value.
- Simplify `.toHaveBeenCalled()` and `.toHaveBeenCalledWith(...)` in all tests.
- I think we need a new test case that verifies that all information is dispalyed, not just the aria attributes. For examples: label, optional radio group indicator, optional radio, error...

File: docs/src/pages/components/cdk-components/radio-group/RadioGroup.jsx [hunk 1]
```diff
@@ -22,6 +27,14 @@ function RadioGroup() {
...
         <Example title="Basic usage" example={basic}></Example>
         <Example title="Controlled Radio Group" example={controlled}></Example>
+        <Example title="Handling errors" example={handlingErrors}>
+          For handling errors, we suggest initializing the <Code>error</Code>{" "}
+          prop with an empty string. This will reserve space for a possible
+          future error message and prevent unintended layout changes. Also, the{" "}
+          <Code>onBlur</Code> event will send <Code>undefined</Code> when there
+          is no error, so you may also need to control this too to avoid the
+          same problem.
+        </Example>
```
[the remark this answers is not in the document]
Comment (GomezIvann) [reply docs/src/pages/components/cdk-components/radio-group/RadioGroup.jsx:37]: After a discussion about this, it was decided to add a bit of text to the example to make it more comprehensible for the user. In the beginning, the whole explanation was in the prop and was too long, making it less accessible from a user's point of view.

File: docs/src/pages/components/cdk-components/radio-group/examples/handlingErrors.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  };
+
+  const options = [
+    { label: "Option A", value: "A" },
+    { label: "Option B", value: "B" },
+    { label: "Option C", value: "C" },
+  ];
+
+  return (
+    <div style={{ display: "flex", flexWrap: "wrap", margin: "36px" }}>
```
[the remark this answers is not in the document]
Comment (GomezIvann) [reply docs/src/pages/components/cdk-components/radio-group/examples/handlingErrors.js:17]: Done!

**Comment by GomezIvann** [comment]
> Comments about tests:
> 
> * _Initial render has correct aria attributes and tabIndex_ case. Add `aria-live` value.
> * Simplify `.toHaveBeenCalled()` and `.toHaveBeenCalledWith(...)` in all tests.
> * I think we need a new test case that verifies that all information is dispalyed, not just the aria attributes. For examples: label, optional radio group indicator, optional radio, error...

I will answer one by one:

- You are right, I have already added it!
- What do you mean by simplify? It's not a random thing when I use one or another, I mean, `toHaveBeenCalledWith` is to ensure the events are not called more than once (which was a regular issue during development with the onFocus) and `toHaveBeenCalled` is for the rest of cases.
- I don't think this will add much value, since this is already been tested at chromatic stories.

**Comment by marcialfps** [comment]
> * What do you mean by 'simplify? It's not a random thing when I use one or another, I mean, `toHaveBeenCalledWith` is to ensure the events are not called more than once (which was a regular issue during development with the onFocus) and `toHaveBeenCalled` is for the rest of cases.

By simplifying I mean that the second function is already checking that the function is called, so you only need to use `toHaveBeenCalledWith(...)`.

If you want to make sure it is called once you should use `toHaveBeenCalledTimes(n)`. 

> * I don't think this will add much value, since this is already been tested at chromatic stories.

I consider chromatic to be the test of the styles, but in this case we are testing the structure of the dom, so we need to verify that we have a label, optional markup...

**Comment by GomezIvann** [comment]
> > * What do you mean by 'simplify? It's not a random thing when I use one or another, I mean, `toHaveBeenCalledWith` is to ensure the events are not called more than once (which was a regular issue during development with the onFocus) and `toHaveBeenCalled` is for the rest of cases.
> 
> By simplifying I mean that the second function is already checking that the function is called, so you only need to use `toHaveBeenCalledWith(...)`.
>
> If you want to make sure it is called once you should use `toHaveBeenCalledTimes(n)`.

I get it, I confused `toHaveBeenCalledWith` with `toHaveBeenCalledTimes(n)` while writing, you are right, I will fix this.

> 
> * I don't think this will add much value, since this is already been tested at chromatic stories.
> 
> I consider chromatic to be the test of the styles, but in this case we are testing the structure of the dom, so we need to verify that we have a label, optional text...

But, I continue without seeing the value it gives just checking it's presence in the DOM with jest. If there were an issue with this, the stories and chromatic would catch it and tell it to us, which is the main purpose of testing...

**Review by marcialfps — APPROVED** [review]

**Pull request merged**